### DISCUSSION TITLE
Per-site category statuses

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#8478] [Ideas]
### Description
I would like to add a feature request: Would you mind to give us an option to enable/disable categories per site?
It is currently possible to enable/disable them with php code via plugin, so we could theoretically overwrite the cp with a custom entry and create a custom controller... but it would be easier if you would just enable the option which requires just a few minutes (I tested it)
In this file [CategoriesController](https://github.com/craftcms/cms/blob/develop/src/controllers/CategoriesController.php#L759) the `populateCategoryFunction` lacks the `enabledForSite`  attribute like the [EntriesController](https://github.com/craftcms/cms/blob/develop/src/controllers/EntriesController.php#L880) has
so changing the CategoriesController to something like

```PHP
    /**
     * Populates an Category with post data.
     *
     * @param Category $category
     *
     * @return void
     */
    private function _populateCategoryModel(Category $category)
    {
        // Set the category attributes, defaulting to the existing values for whatever is missing from the post data
        $category->slug = Craft::$app->getRequest()->getBodyParam('slug', $category->slug);
        $category->enabled = (bool)Craft::$app->getRequest()->getBodyParam('enabled', $category->enabled);
        // edit start
        $category->enabledForSite = (bool)Craft::$app->getRequest()->getBodyParam('enabledForSite', $category->enabled);
        // edit end
        $category->title = Craft::$app->getRequest()->getBodyParam('title', $category->title);

        $fieldsLocation = Craft::$app->getRequest()->getParam('fieldsLocation', 'fields');
        $category->setFieldValuesFromRequest($fieldsLocation);

        // Parent
        $parentId = Craft::$app->getRequest()->getBodyParam('parentId');

        if (is_array($parentId)) {
            $parentId = reset($parentId) ?: null;
        }

        $category->newParentId = $parentId ?: null;
    }
```
would  be much appreciated.

Then the [category template](https://github.com/craftcms/cms/blob/develop/src/templates/categories/_edit.html)  lacks the required field from the [entries edit template](https://github.com/craftcms/cms/blob/develop/src/templates/entries/_edit.html#L178)
```
{{ forms.lightswitch({
     name: 'enabledForSite',
     on:   entry.enabledForSite,
     small: true,
     disabled: isVersion
 }) }}
```
The issue is mentioned here as well [Craft stackexchange](https://craftcms.stackexchange.com/questions/22920/in-craft-3-how-can-i-disable-a-category-on-a-per-site-basis/22923#22923)
So my question is: do we need to create a plugin for that or are you going to include the feature in the future?
### Additional info

- Craft version: Craft Pro 3.0.0-beta.34

**Comment by davist11** [comment]
👍  for this feature request. I imagined setting up the category group would be similar to how the site settings work for Sections

![screen shot 2018-01-19 at 9 56 47 am](https://user-images.githubusercontent.com/199985/35157238-5dcec1ca-fd01-11e7-8183-2fa29a4f0195.png)

**Comment by KatieMFritz** [comment]
+1! In my use case, we have some categories that apply to both sites, and others that are specific. It would be great to be able to enable/disable individual categories on a per-site basis. Or to just have completely separate category groups for each site.

**Comment by boldenamsterdam** [comment]
+1! We have clients who love to see this feature.

**Comment by Saboteur777** [comment]
+1

**Comment by bartrylant** [comment]
+1 I was expecting categories (or better: category groups) to be on a per-site base, but it seems they are not.

**Comment by jsunsawyer** [comment]
To help anyone else who may search for this: multi-site multisite

**Comment by boldenamsterdam** [comment]
+1 Looking forward to this feature!

**Comment by brandonkelly** [comment]
This will come in Craft 3.2 ;)

**Comment by jan-dh** [comment]
Very nice

**Comment by brandonkelly** [comment]
Change of plans… we’re going to hold off on making major localization improvements to categories until 4.0, however we will be getting to #2632 in 3.2, so if you need categories with better localization support, you will be able to do it with a Structure section.

**Comment by kruncher** [comment]
What's the timeline for Craft 4?

**Comment by jsunsawyer** [comment]
@kruncher Based on the recent [Craft Link List](http://craftlinklist.com/issues/88#start) newsletter, it looks like tags and categories are going away in Craft 4 anyway.

In the meantime, you can utilize entries _as_ categories to achieve this functionality, though it might not mesh well with certain fields depending on your current setup.
